*** xmlparse.c-orig	Sun Oct 22 21:47:35 2000
--- xmlparse.c	Tue May 15 19:48:23 2001
***************
*** 2741,2748 ****
  			      parser))
  	return XML_ERROR_NO_MEMORY;
        if (attlistDeclHandler && declAttributeType) {
! 	if (*declAttributeType == '('
! 	    || *declAttributeType == 'N' && declAttributeType[1] == 'O') {
  	  /* Enumerated or Notation type */
  	  if (! poolAppendChar(&tempPool, ')')
  	      || ! poolAppendChar(&tempPool, '\0'))
--- 2741,2749 ----
  			      parser))
  	return XML_ERROR_NO_MEMORY;
        if (attlistDeclHandler && declAttributeType) {
! 	if (    *declAttributeType == '('
! 	    || (*declAttributeType == 'N' && declAttributeType[1] == 'O')) 
! 	{
  	  /* Enumerated or Notation type */
  	  if (! poolAppendChar(&tempPool, ')')
  	      || ! poolAppendChar(&tempPool, '\0'))
***************
*** 2775,2782 ****
  	    && !defineAttribute(declElementType, declAttributeId, declAttributeIsCdata, 0, attVal, parser))
  	  return XML_ERROR_NO_MEMORY;
  	if (attlistDeclHandler && declAttributeType) {
! 	  if (*declAttributeType == '('
! 	      || *declAttributeType == 'N' && declAttributeType[1] == 'O') {
  	    /* Enumerated or Notation type */
  	    if (! poolAppendChar(&tempPool, ')')
  		|| ! poolAppendChar(&tempPool, '\0'))
--- 2776,2784 ----
  	    && !defineAttribute(declElementType, declAttributeId, declAttributeIsCdata, 0, attVal, parser))
  	  return XML_ERROR_NO_MEMORY;
  	if (attlistDeclHandler && declAttributeType) {
! 	  if (    *declAttributeType == '('
! 	      || (*declAttributeType == 'N' && declAttributeType[1] == 'O'))
! 	  {
  	    /* Enumerated or Notation type */
  	    if (! poolAppendChar(&tempPool, ')')
  		|| ! poolAppendChar(&tempPool, '\0'))
